### PR TITLE
chore(lambda): allow the architecture property to be readable from the Function

### PR DESCRIPTION
**Pull request by jcstorms1**
We currently have layers that are designed specifically for arm64 and x86 based functions, and a single stack may have multiple functions with varying architectures. Applying the correct layer to a large list of functions can be a tedious process that may result in applying the incorrect layer to a function. Exposing the architecture at the Function level would allow a user to easily apply the correct layer automatically during deploy by checking the value and pulling the appropriate layer for that architecture. This would eliminate or greatly reduce the possibility of applying the wrong layer type.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jcstorms1** [comment]
Hi @nija-at I was curious if you saw my last comment and if you had any advice on those tests? Thanks!

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 1]
```diff
@@ -672,7 +677,7 @@ export class Function extends FunctionBase {
     if (props.architectures && props.architectures.length > 1) {
       throw new Error('Only one architecture must be specified.');
     }
-    const architecture = props.architecture ?? (props.architectures && props.architectures[0]);
+    const architecture = props.architecture ?? (props.architectures && props.architectures[0]) ?? Architecture.X86_64;
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:680]: Oh no, this is not what I meant by my [previous comment](https://github.com/aws/aws-cdk/pull/17121/files#r736272136). I meant only about fixing the getter. 
The change you have now will cause every CDK user's stack to change (undesirable).

However, I just decided in a [different PR](https://github.com/aws/aws-cdk/pull/17000#discussion_r739116360) to leave such getters as `undefined` if the user has not supplied it. Let's revert this back to returning `undefined` if not specified.

Apologies for the churn.

**Review by nija-at — CHANGES_REQUESTED** [review]

**Comment by jcstorms1** [comment]
@nija-at apologies for the confusion. I've gone ahead and reverted the last two commits as requested and fixed the test assertion.

**Review by nija-at — APPROVED** [review]
Thanks for the updates @jcstorms1

**Comment by jcstorms1** [comment]
Hey @nija-at. I was curious if there was anything else you needed from me to get this merged? I'm hoping to have this feature available soon if possible. Thanks again!

**Comment by nija-at** [comment]
@jcstorms1 - my previous approval was discarded of the manual branch update you triggered. Don't touch the PR once approved. Our bots will do the needed changes and merge.

**Review by nija-at — APPROVED** [review]

**Pull request merged**